android/src/main/kotlin/dev/google/cronet/CronetPlugin.kt

```diff
@@ -0,0 +1,41 @@
+package dev.google.cronet
+
+import androidx.annotation.NonNull
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+import org.chromium.base.ContextUtils
+
+/** CronetPlugin */
+class CronetPlugin: FlutterPlugin, MethodCallHandler {
+  /// The MethodChannel that will the communication between Flutter and native Android
+  ///
+  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+  /// when the Flutter Engine is detached from the Activity
+  private lateinit var channel : MethodChannel
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    System.loadLibrary("cronet.86.0.4240.198")
```

Comment:
Why are we loading these in Kotlin rather than using `DynamicLibrary.open`?

Reply:
If we do not load it using Kotlin first, we get this error -
```
F/chromium(17045): [0418/210141.455792:FATAL:jni_android.cc(96)] Check failed: g_jvm. 
```

Though we're not using Platform Channel from our side in our package itself. Loading the library via Kotlin first then doing `DynamicLibrary.open` on Dart side fixes the issue. (I'm yet to figure out why this is happening though.)

Reply:
Reference: https://github.com/unsuitable001/dart_cronet_sample/issues/3

Reply:
It sounds similar to this: https://github.com/flutter/flutter/issues/73318

Lets just keep it for now and file an issue to look at it later.